lib/src/meta.dart

```diff
@@ -127,3 +112,63 @@ void geomEach(GeoJSONObject geoJSON, GeomEachCallback callback) {
     featureIndex++;
   }
 }
+
+int getFeaturesLength(GeoJSONObject geoJSON) {
+  bool isFeatureCollection =
+      geoJSON.type == GeoJSONObjectTypes.featureCollection;
+  return isFeatureCollection
+      ? (geoJSON as FeatureCollection).features.length
+      : 1;
+}
+
+int getGeometryCollectionLength(GeometryObject geometryObject) {
+  bool isCollection =
+      geometryObject.type == GeoJSONObjectTypes.geometryCollection;
+  return isCollection
+      ? (geometryObject as GeometryCollection).geometries.length
+      : 1;
+}
+
+getFeatureId(GeoJSONObject geoJSON, int index) {
+  bool isFeatureCollection =
+      geoJSON.type == GeoJSONObjectTypes.featureCollection;
+  bool isFeature = geoJSON.type == GeoJSONObjectTypes.feature;
+  return isFeatureCollection
+      ? (geoJSON as FeatureCollection).features[index].id
+      : (isFeature ? (geoJSON as Feature).id : null);
+}
+
+BBox getFeatureBBox(GeoJSONObject geoJSON, int index) {
+  bool isFeatureCollection =
+      geoJSON.type == GeoJSONObjectTypes.featureCollection;
+  bool isFeature = geoJSON.type == GeoJSONObjectTypes.feature;
+  return isFeatureCollection
+      ? (geoJSON as FeatureCollection).features[index].bbox
+      : (isFeature ? geoJSON.bbox : null);
+}
+
+Map<String, dynamic> getFeatureProperties(GeoJSONObject geoJSON, int index) {
+  bool isFeatureCollection =
+      geoJSON.type == GeoJSONObjectTypes.featureCollection;
+  bool isFeature = geoJSON.type == GeoJSONObjectTypes.feature;
+  return isFeatureCollection
+      ? (geoJSON as FeatureCollection).features[index].properties
+      : (isFeature ? (geoJSON as Feature).properties : {});
+}
+
+GeometryObject getGeometryObject(GeoJSONObject geoJSON, int index) {
+  bool isFeatureCollection =
+      geoJSON.type == GeoJSONObjectTypes.featureCollection;
+  bool isFeature = geoJSON.type == GeoJSONObjectTypes.feature;
```

Comment:
it's worth pointing out that we now do this equality test for each iteration of the loop (e.g. for each feature in a feature collection) solely for the purpose of simplifying the interface of these functions.

I do not know the actual performance impacts in dart of doing this though on large feature collections, but wanted to bring this up for discussion anyway.

Reply:
it seems to me that polymorphism is the answer hidden in here somewhere that lets us simplify this `geomEach` function while still doing as few single line operations on each iteration as possible to hyper-optimize performance.

Reply:
we could rewrite it like this, but it still does not feel like a *super huge* improvement. At least we wouldn't need the local variables anymore:
```dart
if(geoJSON is FeatureCollection) {
    return geoJSON.features[index].geometry;
} else if (geoJSON is Feature) {
    return geoJSON.geometry;
} else {
    return geoJSON;
}
```

or in a shorter form:
```dart
return geoJSON is FeatureCollection
      ? geoJSON.features[index].geometry
      : (geoJSON is Feature ? geoJSON.geometry : geoJSON);
```

Reply:
I think general what I was thinking might make this "weird logic" taste better is splitting the for loop up into more functions that geomEach calls, e.g. `_forEachGeomInFeatureCollection`, `_forEachGeomInFeature`, and `_forEachGeomInGeometryObject` in which case I wouldn't need to perform the "is this a feature/featurecollection" checks on each feature in large feature datasets, the `_forEachGeomInFeatureCollection` would be written specifically for the `isFeatureCollection` use case, etc etc